src/components/pill-selector-dropdown/RoundPill.js

```diff
@@ -48,31 +51,19 @@ class RoundPill extends React.PureComponent<Props, State> {
     isMounted: boolean = false;
 
     getStyles = (): string => {
-        const { className, isSelected, isDisabled, hasWarning, isValid } = this.props;
+        const { className, isSelected, isDisabled, isValid, variant } = this.props;
+        // Invalid pills are always displayed as red regardless of which variant is selected
+        const activeVariant = isValid ? variant : PILL_VARIANT_DEFAULT;
 
         return classNames('bdl-RoundPill', className, {
             'bdl-RoundPill--selected': isSelected && !isDisabled,
             'bdl-RoundPill--disabled': isDisabled,
-            'bdl-RoundPill--warning': hasWarning,
+            'bdl-RoundPill--warning': activeVariant === PILL_VARIANT_WARNING,
+            'bdl-RoundPill--waived': activeVariant === PILL_VARIANT_WAIVED,
             'bdl-RoundPill--error': !isValid,
         });
     };
 
-    getPillType = (): ?string => {
```

Comment:
`RoundPill` and `LabelPill` styles have diverged a little, so it's no longer necessary to set the specific type on the label pill (i.e., `RoundPill` styles are overrides of the default `LabelPill` style.

Reply:
I think they shouldn't conflict because `RoundPill` controls the rendering of `LabelPill` so it should always be set to default now.